src/database/database.rs

```diff
@@ -148,6 +158,67 @@ impl Database {
         Err(self.connection.unable_to_connect_error())
     }
 
+    #[cfg(feature = "sync")]
+    pub(super) fn run_failsafe<F, R>(&self, task: F) -> Result<R>
+    where
+        F: Fn(ServerDatabase, ServerConnection, bool) -> Result<R>,
+    {
```

Comment:
Higher-order functions can't be as easily converted from async to sync, even with type aliases. Fortunately, we don't have too many of those.